iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -193,7 +208,7 @@ constexpr inline bool optional<T>::operator!=(const optional<T>& rhs) const noex
 }
 
 template <typename T>
-constexpr inline bool optional<T>::operator!=(const nullopt_t& rhs) const noexcept
+constexpr inline bool optional<T>::operator!=(const nullopt_t rhs) const noexcept
```

Comment:
Please get rid of `rhs`.

Reply:
@elfenpiff then clang-tidy will complain. The agreed solution was to use `IOX_MAYBE_UNUSED`

Reply:
![intro-import](https://user-images.githubusercontent.com/56729169/189874936-214af21c-26b3-45ff-a317-a411f51f3278.jpg)

There is always that one tool.